src/main/java/com/google/firebase/messaging/Aps.java

```diff
@@ -159,6 +203,13 @@ public Builder setThreadId(String threadId) {
       return this;
     }
 
+    /**
+     * Builds a new {@link Aps} instance from the fields set on this builder.
+     *
+     * @return A non-null {@link Aps}.
+     * @throws IllegalArgumentException If the alert is specified both as an object and a string.
+     *     Or if the same field is set both using a setter method, and as a custom field.
```

Comment:
Is this last part of this enforced somewhere?

Reply:
`ImmutableMap.Builder` checks for this internally. We have a test case to verify.

Reply:
Got it. I saw the test cases but I was wondering if maybe it caught an unrelated `IllegalArgumentException`.